### PR TITLE
HDDS-3142. Create isolated enviornment for OM to test it without SCM.

### PR DESCRIPTION
**Pull request by elek**
## What changes were proposed in this pull request?

`OmKeyGenerator` class from Freon can generate keys (open key + commit key). But this test tests both OM and SCM performance. It seems to be useful to have a method to test only the OM performance with faking the response from SCM.

Can be done easily with the same approach what we have in HDDS-3023: A simple utility class can be implemented and with byteman we can replace the client calls with the fake method.

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-3142

## How was this patch tested?

 1. Download [byteman](https://byteman.jboss.org/)
 2. Start a pure OM (`ozone om --init` + `ozone om`) with the following JVM parameters: (change the path) 

```
-javaagent:/home/elek/prog/byteman/lib/byteman.jar=script:/home/elek/projects/ozone/dev-support/byteman/mock-scm.btm,boot:/home/elek/prog/byteman/lib/byteman.jar
-Dorg.jboss.byteman.transform.all
``` 

 3. Start a simple freon test: `ozone freon omkg`

Expected result: It should be possible to init and start OM without SCM and test it with the key generator)

### CODE REVIEW

File: hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/freon/FakeScmBlockLocationProtocolClient.java [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+/**
+ * Fake SCM client to return a simulated block location.
+ */
+public final class FakeScmBlockLocationProtocolClient {
+
+  private static final Logger LOGGER =
+      LoggerFactory.getLogger(FakeScmBlockLocationProtocolClient.class);
+
+  public static final int BLOCK_PER_COUNTER = 1000;
```
Comment (adoroszlai) [inline hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/freon/FakeScmBlockLocationProtocolClient.java]: Intended to be `BLOCK_PER_CONTAINER`?

File: hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/freon/FakeScmContainerLocationProtocolClient.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+  public static ScmContainerLocationResponse submitRequest(
+      ScmContainerLocationRequest req)
+      throws IOException {
+    try {
+      if (req.getCmdType() == Type.QueryNode) {
+        Builder builder = NodeQueryResponseProto.newBuilder()
+            .addDatanodes(Node.newBuilder()
+                .setNodeID(FakeClusterTopology.INSTANCE.getDatanode(1))
+                .addNodeStates(NodeState.HEALTHY)
+                .build());
```
Comment (adoroszlai) [inline hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/freon/FakeScmContainerLocationProtocolClient.java]: Isn't node 1 added in the loop, too?

File: hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/freon/FakeClusterTopology.java [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+public class FakeClusterTopology {
+
+  private static final Logger LOGGER =
+      LoggerFactory.getLogger(FakeClusterTopology.class);
+
+  public static final FakeClusterTopology INSTANCE = new FakeClusterTopology();
+
+  private List<DatanodeDetailsProto> datanodes = new ArrayList<>();
+
+  private List<Pipeline> piplines = new ArrayList<>();
```
Comment (adoroszlai) [inline hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/freon/FakeClusterTopology.java]: typo: `piplines` -> `pipelines`

File: hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/freon/FakeScmContainerLocationProtocolClient.java (hunk 2 above)
Comment (elek) [reply hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/freon/FakeScmContainerLocationProtocolClient.java]: Yeah, it came from an old code, thanks.

**Comment by elek** [comment]
Thanks the review @adoroszlai, I fixed the typo bugs.

> Would be nice to include mock-scm.btm script in hadoop-ozone/dist/target for testing.

100% agree. I created an issue to track this: https://issues.apache.org/jira/browse/HDDS-3232

**Comment by adoroszlai** [comment]
Thanks @elek for updating the patch.

**Comment by elek** [comment]
HDDS-3160 is accidentally added to this pr (sorry for that)
 1. I will revert the original merge
 2. Remove HDDS-3160
 3. Create a new PR without that
 4. And merge it after a green build

**Pull request merged**